### PR TITLE
Add guidelines for commit messages and branch selection

### PR DESCRIPTION
**Pull request by flihp**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -21,6 +21,12 @@ consider this the window for comments.
 * All tests must pass on Travis CI for the merge to occur.
 * All changes must not introduce superfluous changes or whitespace errors.
 * All changes should adhere to the coding standard.
+* All commits should adhere to the git commit message guidlines described
+here: https://chris.beams.io/posts/git-commit/ with the following exceptions.
```
Comment (martinezjavier) [inline CONTRIBUTING.md]: Maybe it's better to refer to https://www.kernel.org/doc/html/latest/process/submitting-patches.html#the-canonical-patch-format which is a more formal and summarized version of what's mentioned in that article?

Or do you have any issues with the canonical patch format used by Linux?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -21,6 +21,12 @@ consider this the window for comments.
...
+ * We allow commit subject lines less than 80 characters.
```
Comment (martinezjavier) [inline CONTRIBUTING.md]: Did you mean s/less/more ? Otherwise I'm not sure I follow this sentence.

Comment (flihp) [reply CONTRIBUTING.md]: haha yeah whoops, long day

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -21,6 +21,12 @@ consider this the window for comments.
...
+ * Commit subject lines should be prefixed with a string identifying the
+ effected subsystem / object. If the change is spread over a number of
+subsystems them the prefix may be omitted.
```
Comment (martinezjavier) [inline CONTRIBUTING.md]: Do you also want to require contributors to sign-off their commits? If so I would also mention here and add a reference to https://developercertificate.org/

File: CONTRIBUTING.md (hunk 1 above)
Comment (flihp) [reply CONTRIBUTING.md]: I ran across both but the `canonical-patch-format` includes a lot more data than just commit message format and describes everything in the context of sending patches to a mailing list. I figured including all of that would probably confuse folks.

Comment (martinezjavier) [reply CONTRIBUTING.md]: @flihp fair enough.

File: CONTRIBUTING.md (hunk 3 above)
Comment (flihp) [reply CONTRIBUTING.md]: that's a good idea

**Comment by flihp** [comment]
Appreciate your feedback as always @martinezjavier.

**Pull request merged**